src/js/libcs/OpSound.js

```diff
@@ -1,48 +1,533 @@
+                }
+                for (var i = 0; i < harmonics.length; i++) {
+                    if (i < lines[line].oscNodes.length) {
+                        lines[line].oscNodes[i] = playOscillator(line, partials[i].value.real * (i + 1) * freq, harmonics[i].value.real);
+                    } else {
+                        lines[line].oscNodes.push(playOscillator(line, partials[i].value.real * (i + 1) * freq, harmonics[i].value.real));
+                    }
+                }
+                lines[line].masterGain.gain.linearRampToValueAtTime(amp, audioCtx.currentTime + release + attack);
+
+                if (damp > 0) {
+                    lines[line].masterGain.gain.setTargetAtTime(0.0, audioCtx.currentTime + release + attack, (1 / damp));
+                } else if (damp < 0) {
+                    lines[line].masterGain.gain.setTargetAtTime(1, audioCtx.currentTime + release + attack, (-damp));
+                }
+            } else { //restart -> false so just change damp & timbre
+                for (var i = 0; i < harmonics.length; i++) {
+                    lines[line].oscNodes[i].oscNode.frequency.value = partials[i].value.real * (i + 1) * freq;
+                    lines[line].oscNodes[i].gainNode.gain.value = harmonics[i].value.real;
+                }
+                if (damp > 0) {
+                    lines[line].masterGain.gain.setTargetAtTime(0.0, audioCtx.currentTime, (1 / damp));
+                } else if (damp < 0) {
+                    lines[line].masterGain.gain.setTargetAtTime(1, audioCtx.currentTime, (-damp));
                 }
             }
         }
     }
+    return nada;
+};
 
 
-    var v0 = evaluateAndVal(args[0]);
-    var linenumber = 0;
-    if (v0.ctype === 'number') {
-        handleModifs();
-        var lines = sound.lines;
-        var f = v0.value.real;
-        if (lines[linenumber] === 0) {
-            // Was bitte sollte die Funktion T an dieser Stelle sein?
-            // lines[linenumber]=T("sin", {freq:f,mul:0.6}).play();
+evaluator.playsin$0 = function(args, modifs) {
+    var erg;
+    if (modifs.line !== undefined) {
+        erg = evaluate(modifs.line);
+        if (erg.ctype === 'number') {
+            var val = erg.value.real;
+        } else if (erg.ctype === 'string') {
+            var val = erg.value;
+        } else {
+            var val = 0;
+        }
+        if (val === 0) {
+            var line = 0;
+        } else if (!linesDict[val]) {
+            return nada;
+        } else {
+            var line = linesDict[val];
+        }
+        evaluator.playsin$1([CSNumber.real(lines[line].oscNodes[0].oscNode.frequency.value)], modifs);
 
+    }
+    return nada;
+};
 
-        } else {
-            lines[linenumber].set({
-                freq: f
-            });
+
+evaluator.playfunction$1 = function(args, modifs) {
+    if (!audioCtx) {
+        audioCtx = getAudioContext();
+    }
+
+    function handleModifs() {
```

Comment:
maybe you could make `handleModifs` one level more global and reuse the old one from above (avoid redundancy)